Cookbook/Technical-Documents/DebugBuilder.md

```diff
@@ -0,0 +1,96 @@
+Debug Builder
+======================
+
+[\[CNSMR\-1602\] Add \`DebugAuthenticatedRootBuilder\` for faster UI debugging iteration by inamiy · Pull Request \#7645 · Babylonpartners/babylon\-ios](https://github.com/Babylonpartners/babylon-ios/pull/7645)
+
+Above PR introduces `DebugAuthenticatedRootBuilder` that allows to directly show a target view builder after authentication.
+
+This feature improves overall UI debugging iteration speed.
+
+## How to make a debug-builder
+
+1. Add a new case in `enum DebugBuilderPattern`
+2. Create a (mock) builder in `DebugAuthenticatedRootBuilder`
+3. Update `Settings.bundle/Root.plist` by adding a new case key
+4. Set `Settings.bundle`'s userDefaults flag via `Settings.app`
```

Comment:
```suggestion
4. Build and launch your app at least once after those changes, so that the Settings app can now discover the new values in the `Root.plist`
4. Set `Settings.bundle`'s userDefaults flag via `Settings.app`
```

(Note that you could also use `-debugBuilder YourCaseName` in your launch parameters in your Scheme to override those User Defaults at launch time. But might not be a good idea to suggest that since people risk forgetting to remove those and they would end up being commited)